### ISSUE TITLE
rto-llama's dev-dependency on llama-cpp-2 is not optional, so every --all-targets build compiles llama.cpp

### ISSUE THREAD

**Issue by pixie79** [OffeneDatenmodellierung/Roteiro#483]
`crates/rto-llama/Cargo.toml`:

```toml
[dependencies]
# Opt-in: pulls a C/C++ toolchain (cmake, clang, libclang) to build vendored llama.cpp.
llama-cpp-2 = { version = "0.1.154", optional = true }   # line 19 — correctly optional

[dev-dependencies]
# ... Inert unless a test that uses it is compiled.
llama-cpp-2 = { version = "0.1.154" }                    # line 29 — NOT optional
```

**Dev-dependencies are not feature-gated.** So `cargo test` / `cargo clippy --all-targets` compiles vendored llama.cpp at *every* feature set, including `--no-default-features`, and needs `cmake clang libclang-dev` to do it.

The comment — *"Inert unless a test that uses it is compiled"* — is true of **linking** and false of **building**. That is the contract-drift class: a stated property the mechanism does not have. `cargo build` and `cargo install` are genuinely unaffected, so users never pay it; CI pays it on every job, forever.

## Found twice, filed neither time

First by the worker on #364, which correctly scoped it out and documented it instead. Now again by #482, which hit it while measuring the new `no-default-features` job and had to correct my estimate:

> This job is *not* cheaper than the ones already running. `rto-llama` takes `llama-cpp-2` as an unconditional dev-dependency, so `--all-targets` builds vendored llama.cpp at every feature set.

I asserted on #445 that the job *"compiles strictly less than the jobs already running (no llama.cpp at this feature set)"*. That was wrong, and this is why.

## The cost, concretely

Three CI jobs — `checks`, `default-features`, `no-default-features` — each install a C/C++ toolchain and build a vendored C++ tree that only `tests/batch_numerics.rs` uses. `msrv` too, on `--all-targets`.

## Two directions, neither obviously right

1. **Make it optional and gate the consumer.** `tests/batch_numerics.rs` exists to compare llama.cpp's logits at two batch widths — a question about llama.cpp that `LlamaEngine`'s API deliberately does not expose (it is the evidence behind Stage 30's finding that speculative decoding changes output). Gating it means that test runs only under `--all-features`, which it arguably already needs a model for anyway. Establish whether it can run at all without one before assuming this is free.
2. **Correct the comment** and accept the cost, so at least the file stops asserting something untrue.

Doing neither is the one option to rule out: the claim has now misled two workers and one orchestrator.
